packages/swingset-liveslots/test/test-handled-promises.js

```diff
@@ -0,0 +1,294 @@
+/* eslint-disable no-await-in-loop, @jessie.js/no-nested-await */
+import test from 'ava';
+import '@endo/init/debug.js';
+
+import { Far } from '@endo/marshal';
+import { Fail } from '@agoric/assert';
+import { M } from '@agoric/store';
+import { makePromiseKit } from '@endo/promise-kit';
+// import { makeStoreUtils } from '../../vat-data/src/vat-data-bindings.js';
+import { makeExoUtils } from '../../vat-data/src/exo-utils.js';
```

Comment:
This import feels problematic. At the very least, let's change it to `@agoric/vat-data/src/exo-util.js` (using `../..` is kinda cheating / assuming the monorepo layout, and I'm likely to be moving swingset-liveslots out to a separate repository, which will break `../../`)

At a deeper level, Exos are not a liveslots concern. Liveslots provides Kinds, and userspace can use `@agoric/vat-data` to get Exos on top of that. We need to make sure that liveslot's `defineKind` works when used by `makeExo` stuff, but I think that should be the responsibility of `vat-data` rather than liveslots.

And I'm worried about the coupling this induces (especially because of my upcoming #6596 work): currently swingset-liveslots has a `devDependency` upon vat-data solely to get the types for `DefineKindOptions` and `DurableKindHandle`, and I think those types should be *exported* by liveslots rather than having liveslots *import* them from vat-data. With that change, we could shed the liveslots -> vat-data `devDependency` entirely, which is going to help with the #6596 packaging effort (the build process for `xsnap-supervisor`, which imports liveslots, can happen in a workspace/repo/`node_modules` that doesn't have swingset at all, which would prune a cycle).

Could you have this test work with a Kind instead of an Exo, and remove the dependency on `vat-data`?

Reply:
Huh, I thought I had already done that. Fixed.